File: backend/app/schemas/coach.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class CoachFeedback(BaseModel):
    available: bool
    message: str | None = None


class PhotoFeedbackRequest(BaseModel):
    image_base64: str = Field(min_length=1)
    media_type: str = "image/jpeg"


class PracticeSessionFeedbackRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    practice_session_id: int
    kind: str
    message: str
    photo_url: str | None
    created_at: datetime

File: src/api/coach.ts
import { api } from '@/api/client';

export type CoachFeedback = {
  available: boolean;
  message: string | null;
};

export type PracticeSessionFeedbackEntry = {
  id: number;
  practice_session_id: number;
  kind: 'text' | 'photo';
  message: string;
  photo_url: string | null;
  created_at: string;
};

export const getPracticeSessionFeedback = (practiceSessionId: number) =>
  api.get<CoachFeedback>(`/coach/practice-sessions/${practiceSessionId}/feedback`);

export const getPracticeSessionPhotoFeedback = (
  practiceSessionId: number,
  input: { image_base64: string; media_type: string },
) => api.post<CoachFeedback>(`/coach/practice-sessions/${practiceSessionId}/photo-feedback`, input);

export const getPracticeSessionFeedbackHistory = (practiceSessionId: number) =>
  api.get<PracticeSessionFeedbackEntry[]>(`/coach/practice-sessions/${practiceSessionId}/history`);

export const getFeedbackHistory = () => api.get<PracticeSessionFeedbackEntry[]>('/coach/history');
